internal/parser/reparser.go

```diff
@@ -0,0 +1,88 @@
+package parser
+
+import (
+	"github.com/microsoft/typescript-go/internal/ast"
+	"github.com/microsoft/typescript-go/internal/core"
+)
+
+type jsDeclarationKind int
+
+const (
+	jsDeclarationKindNone jsDeclarationKind = iota
+	/// module.exports = expr
+	jsDeclarationKindModuleExports
+	/// exports.name = expr
+	/// module.exports.name = expr
+	jsDeclarationKindExportsProperty
+	/// className.prototype.name = expr
+	jsDeclarationKindPrototypeProperty
```

Comment:
I included the last 3 entries for future use, although I'm not 100% sure that they will be needed, or stay private to this module.